src/list.alp

```diff
@@ -0,0 +1,307 @@
+{-| A library for manipulating lists of values.
+-}
+
+module list
+
+export head/1, tail/1, isEmpty/1, map/2, any/2, all/2, reverse/1, length/1,
+       concat/1, filter/2, append/2, foldLeft/3, foldRight/3, sort/1
+
+import_type option.option
+
+
+{-| Extract the first element of a list
+
+      head [3, 4] == (Some 3)
+      head [] == (None)
+-}
+val head 'a : fn (list 'a) -> option 'a
+let head xs =
+  match xs with
+  | [] -> None
+  | x :: _ -> Some x
+
+test "head of list" =
+  assert.equal (head [3, 4]) (Some 3)
+
+test "head of empty list" =
+  assert.equal (head []) (None)
+
+
+{-| Extract the rest of the list.
+
+      tail [1,2,3] == (Some [2,3])
+      tail [] == (None)
+-}
+-- val tail 'a : fn (list 'a) -> option (list 'a) -- TODO: Why doesn't it compile with this typesig?
```

Comment:
Could you please file a bug against Alpaca with a minimal module example?  Seems like something we should fix in 0.2.9.  Thanks!